src/Microsoft.AspNetCore.Razor.Language/Legacy/TagHelperBlockRewriter.cs

```diff
@@ -715,39 +360,289 @@ private static string GetPropertyType(string name, IEnumerable<TagHelperDescript
             return firstBoundAttribute;
         }
 
-        private static bool IsQuote(SyntaxToken token)
+        private static string GetAttributeValueContent(RazorSyntaxNode attributeBlock)
         {
-            return token.Kind == SyntaxKind.DoubleQuote ||
-                   token.Kind == SyntaxKind.SingleQuote;
+            if (attributeBlock is MarkupTagHelperAttributeSyntax tagHelperAttribute)
+            {
+                return tagHelperAttribute.Value?.GetContent();
+            }
+            else if (attributeBlock is MarkupAttributeBlockSyntax attribute)
+            {
+                return attribute.Value?.GetContent();
+            }
+
+            return null;
         }
 
-        private static void ConfigureNonStringAttribute(SpanBuilder builder, bool isDuplicateAttribute)
+        private class AttributeValueRewriter : SyntaxRewriter
```

Comment:
Whatever is in it is probably ugly. TagHelperBlockRewriter used to do a whole bunch of "special" rewriting of attribute values. Keeping the same behavior while operating on the new tree makes it really ugly. I plan to clean this up as much as possible as part of attribute value syntax improvements.